pkg/networkserver/mac/utils.go

```diff
@@ -222,6 +222,32 @@ func DeviceDefaultPingSlotPeriodicity(dev *ttnpb.EndDevice, defaults ttnpb.MACSe
 	}
 }
 
+func DeviceUplinkDwellTime(dev *ttnpb.EndDevice, phy *band.Band, defaults ttnpb.MACSettings) *ttnpb.BoolValue {
+	switch {
+	case dev.GetMacSettings().GetUplinkDwellTime() != nil:
+		return &ttnpb.BoolValue{Value: dev.MacSettings.UplinkDwellTime.Value}
+	case defaults.UplinkDwellTime != nil:
+		return &ttnpb.BoolValue{Value: defaults.UplinkDwellTime.Value}
+	case phy.BootDwellTime.Uplinks != nil:
+		return &ttnpb.BoolValue{Value: *phy.BootDwellTime.Uplinks}
+	default:
+		return nil
+	}
+}
+
+func DeviceDownlinkDwellTime(dev *ttnpb.EndDevice, phy *band.Band, defaults ttnpb.MACSettings) *ttnpb.BoolValue {
+	switch {
+	case dev.GetMacSettings().GetDownlinkDwellTime() != nil:
+		return &ttnpb.BoolValue{Value: dev.MacSettings.DownlinkDwellTime.Value}
+	case defaults.DownlinkDwellTime != nil:
+		return &ttnpb.BoolValue{Value: defaults.DownlinkDwellTime.Value}
+	case phy.BootDwellTime.Downlinks != nil:
+		return &ttnpb.BoolValue{Value: *phy.BootDwellTime.Downlinks}
+	default:
+		return nil
+	}
+}
```

Comment:
So, in case of 1.0.2 rev. B with no per-device setting, this method returns `nil`. Then, only if the frequency plan has dwell time set (true or false), it will lead to a `TxParamSetupReq` per https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.17/pkg/networkserver/mac/tx_param_setup.go#L47-L56. Right?

Doesn't this mean that all frequency plans in bands where dwell time may be applicable, need to have a dwell time set explicitly?

Currently, in our general plans like `AS_923`, `AS_920_923` and `AS_923_925`, we don't set dwell time. That means that `TxParamSetupReq` is not sent.